2016/exercises/final_project/Alba-Petar-Rebecca-Simon/PolytopeProject/Matrix.cpp

```diff
@@ -0,0 +1,245 @@
+//
+//  Matrix.cpp
+//  PolytopeProject
+//
+
+
+#include "Matrix.hpp"
+
+Matrix::Matrix(vector<Point> points){
+    for(int i=0; i<points[0].get_coordinates().size(); i++){
+        matrix.push_back(vector<mpq_class>());
+    }
+    
+    for(int i=0; i<points.size(); i++){
+        vector<mpq_class> coord = points[i].get_coordinates();
+        for(int j=0; j<coord.size(); j++){
+            coord[j].canonicalize();
```

Comment:
Oh, when reading the documentation: `All arithmetic operators require their operands in canonical form, and will return results in canonical form.` I must have missed the second part of the sentence.